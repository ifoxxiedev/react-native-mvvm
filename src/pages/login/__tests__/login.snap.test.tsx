import {it, describe, expect, jest} from '@jest/globals';
import renderer from 'react-test-renderer';
import LoginView from '../view';
import {LoginViewModel} from '../models';

jest.mock('../view.model', () => {
  const fnLoginViewModel = jest.fn();
  fnLoginViewModel.mockReturnValue({
    isLoading: false,
    password: '',
    email: '',
    onSubmit: jest.fn(),
    setEmail: jest.fn(),
    setPassword: jest.fn(),
  } as LoginViewModel);
  return fnLoginViewModel;
});

describe('Login snapshot test', () => {
  it('should match snapshot', () => {
    const {toJSON} = renderer.create(<LoginView />);
    expect(toJSON()).toMatchSnapshot();
  });
});
