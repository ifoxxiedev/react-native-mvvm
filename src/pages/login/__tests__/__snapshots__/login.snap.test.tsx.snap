// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot test should match snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Text>
      E-mail
    </Text>
    <TextInput
      autoCapitalize="none"
      autoComplete="off"
      onChangeText={[MockFunction]}
      placeholder="email@email.com"
      value=""
    />
  </View>
  <View>
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Password"
      secureTextEntry={true}
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Login
    </Text>
  </View>
</RCTSafeAreaView>
`;
